### PR TITLE
Feature to Mark an Entire Job's Parameters as Sensitive

### PR DESCRIPTION
**Pull request by timizuoebideri1**
# Closes: #2091 
# What's Changed

1. Added sensitive to `BaseJob` (as `@classproperty`  which defaults to `False`), `JOB_OVERRIDABLE_FIELDS`,  `JobFilterSet`, `JobFilterForm`, `Job` model (also added sensitive_override) and `JobTable`
2. Added validation to the form and serializer to prevent setting a job to both `sensitive` and `approval_required`.
3. Restrict/raise exceptions if starting a sensitive job also marked for approval.(UI/API)
4. Sensitive jobs can only be scheduled immediately (UI/API) and hide Job Execution form in UI if the job is sensitive.
5. Sensitive jobs cant be rerun

Not Done:

- Raise exception on classes that inherit from `BaseJob` if both` Meta.approval_required` and `Meta.sensitive` is set to `True` because this prevents nautobot from starting up.

## Screenshots
**Sensitive jobs marked for approval cannot start**
Screenshot 2022-07-29 at 8 05 05 AM

Screenshot 2022-07-29 at 8 08 36 AM

**Sensitive jobs cannot rerun**
Screenshot 2022-07-29 at 8 24 59 AM

**Error message displayed if job is sensitive and also requires approval**
Screenshot 2022-07-29 at 8 09 28 AM

Screenshot 2022-07-29 at 8 27 50 AM

**Job Execution panel hidden if sensitive job is immediate**
Screenshot 2022-07-29 at 9 17 44 AM

# TODO

- [ ] Explanation of Change(s)
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/extras/api/serializers.py [hunk 1]
```diff
@@ -643,6 +643,7 @@ class Meta:
             "description_override",
             "installed",
             "enabled",
+            "sensitive",
```
Comment (glennmatthews) [inline nautobot/extras/api/serializers.py:646]: I wonder if something like `sensitive_variables` would be clearer throughout?

File: nautobot/extras/tables.py [hunk 2]
```diff
@@ -69,7 +69,7 @@
 """
 
 JOB_BUTTONS = """
-<a href="{% url 'extras:job_run' slug=record.slug %}" class="btn btn-primary btn-xs" title="Run/Schedule" {% if not perms.extras.run_job or not record.enabled or not record.installed %}disabled="disabled"{% endif %}><i class="mdi mdi-play" aria-hidden="true"></i></a>
+<a href="{% url 'extras:job_run' slug=record.slug %}" class="btn btn-primary btn-xs" title="Run/Schedule" {% if not perms.extras.run_job or not record.enabled or record.sensitive and record.approval_required or not record.installed %}disabled="disabled"{% endif %}><i class="mdi mdi-play" aria-hidden="true"></i></a>
```
Comment (glennmatthews) [inline nautobot/extras/tables.py:72]:
```suggestion
<a href="{% url 'extras:job_run' slug=record.slug %}"
    class="btn btn-primary btn-xs" title="Run/Schedule"
    {% if not perms.extras.run_job or not record.enabled or record.sensitive and record.approval_required or not record.installed %}disabled="disabled"{% endif %}
><i class="mdi mdi-play" aria-hidden="true"></i></a>
```

Though honestly I'm not sure this check is needed - we should never get into this state, right?

File: nautobot/extras/templates/extras/job.html [hunk 3]
```diff
@@ -29,6 +29,11 @@ <h1>{{ job_model }}</h1>
         <div role="tabpanel" class="tab-pane active" id="run">
             <div class="row">
                 <div class="col-md-6 col-md-offset-3">
+                    {% if job_model.sensitive and job_model.approval_required %}
+                        <div class="alert alert-danger">
+                            <i class="mdi mdi-alert"></i> This job is sensitive and can not be marked for approval.
+                        </div>
+                    {% endif %}
```
Comment (glennmatthews) [inline nautobot/extras/templates/extras/job.html:36]: Similarly, this shouldn't happen right?

File: nautobot/extras/templates/extras/job.html [hunk 4]
```diff
@@ -105,11 +118,12 @@ <h1>{{ job_model }}</h1>
...
+    var sensitive = {{ job_model.sensitive|yesno:"true,false" }};
     var requires_approval = {{ job_model.approval_required|yesno:"true,false" }};
     var run_now_text;
     var schedule_text;
 
-    if (requires_approval) {
+    if (requires_approval && !sensitive) {
         run_now_text = "Request to Run Job Now";
         schedule_text = "Request to Schedule Job";
     } else {
```
Comment (glennmatthews) [inline nautobot/extras/templates/extras/job.html:129]: Perhaps rethink this approach to templating? Something along the lines of:

```django
{% if job_model.approval_required and not job_model.sensitive %}
run_now_text = "Request to Run Job Now";
schedule_text = "Request to Schedule Job";
{% else %}
run_now_text = "Run Job Now";
schedule_text = "Schedule Job";
{% endif %}
```

File: nautobot/extras/templates/extras/job.html [hunk 5]
```diff
@@ -147,8 +161,14 @@ <h1>{{ job_model }}</h1>
...
-        $("#id__schedule_type").change(toggleExecutionType);
-        toggleExecutionType();
+        if ("{{ job_model.sensitive }}" == "False") {
+            $("#id__schedule_type").change(toggleExecutionType);
+            toggleExecutionType();
+        }
+        else {
+            $("#id__schedule_type").val("immediately");
+            $(".job_execution").hide();
+        }
```
Comment (glennmatthews) [inline nautobot/extras/templates/extras/job.html:171]:
```suggestion
        {% if not job_model.sensitive %}
        $("#id__schedule_type").change(toggleExecutionType);
        toggleExecutionType();
        {% else %}
        $("#id__schedule_type").val("immediately");
        $(".job_execution").hide();
        {% endif %}
```

**Comment by timizuoebideri1** [comment]
opened a new PR: https://github.com/nautobot/nautobot/pull/2125

**Pull request closed without merging**